tests-functional/tests/test_wakuext_community_archives.py

```diff
@@ -0,0 +1,107 @@
+        # Define codex as archive distribution preference
+        self.another_member.wakuext_service.set_archive_distribution_preference("codex")
+
+        self.fake_address = "0x" + str(uuid4())[:8]
+        self.community_id = self.create_community(self.creator, historyArchiveSupportEnabled=True)
+
+        # Ensure that no community archive exists initially
+        has_archive = self.creator.wakuext_service.has_community_archive(self.community_id)
+        assert has_archive is False, "Creator should not have community archive initially"
+        has_archive = self.member.wakuext_service.has_community_archive(self.community_id)
+        assert has_archive is False, "Member should not have community archive initially"
+        has_archive = self.another_member.wakuext_service.has_community_archive(self.community_id)
+        assert has_archive is False, "Another member should not have community archive initially"
+
+        # Connect members to community codex client
+        # In the real life, this would be done via discovery
+        info = self.creator.wakuext_service.debug()
+        self.member.wakuext_service.connect(info["id"], info["addrs"])
+        self.another_member.wakuext_service.connect(info["id"], info["addrs"])
+
+        self.display_name = "chat_" + str(uuid4())
+        self.chat_payload = {
+            "identity": {
+                "displayName": self.display_name,
+                "emoji": "😀",
+                "color": "#1f2c75",
+                "description": self.display_name,
+            },
+            "viewersCanPostReactions": False,
+            "hideIfPermissionsNotMet": False,
+            "permissions": {"access": 1},
+        }
+
+    def test_community_archive_index_exists(self):
+        # Create community chat
+        create_resp = self.creator.wakuext_service.create_community_chat(self.community_id, self.chat_payload)
+        chat_id = create_resp.get("chats")[0].get("id")
+
+        # Wait for member to receive chat creation signal
+        self.join_community(member=self.member, admin=self.creator)
+        self.member.find_signal_containing_pattern(SignalType.MESSAGES_NEW.value, event_pattern=chat_id, timeout=10)
+
+        # Send a message to the community chat
+        text = f"Hi @{self.member.public_key}"
+        send_resp = self.creator.wakuext_service.send_chat_message(chat_id, text)
+        assert send_resp.get("chats")[0].get("lastMessage").get("text") == text
+        message_id = send_resp.get("messages", [])[0].get("id", "")
+
+        # Wait for member to receive the new message
+        self.member.find_signal_containing_pattern(SignalType.MESSAGES_NEW.value, event_pattern=message_id, timeout=10)
+        member_msgs_resp = self.member.wakuext_service.chat_messages(chat_id)
+        assert member_msgs_resp.get("messages")[0].get("text") == text
+
+        # Just make sure that archive are generated
+        time.sleep(10)
```

Comment:
Might be a good idea to pull this out into a variable, so that a change in the
profile `message_archive_interval=10` setting does not cause the test to fail in
an odd way. It also self-describes the reason for sleeping here.